src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -157,15 +204,220 @@ EscrowCreate::preflight(PreflightContext const& ctx)
     return preflight2(ctx);
 }
 
+template <ValidIssueType T>
+static TER
+escrowPreclaimHelper(
+    PreclaimContext const& ctx,
+    AccountID const& issuer,
+    AccountID const& account,
+    AccountID const& dest,
+    STAmount const& amount);
+
+template <>
+TER
+escrowPreclaimHelper<Issue>(
+    PreclaimContext const& ctx,
+    AccountID const& issuer,
+    AccountID const& account,
+    AccountID const& dest,
+    STAmount const& amount)
+{
+    // If the issuer is the same as the account, return tesSUCCESS
+    if (issuer == account)
+        return tesSUCCESS;
+
+    // If the lsfAllowTokenLocking is not enabled, return tecNO_PERMISSION
+    auto const sleIssuer = ctx.view.read(keylet::account(issuer));
+    if (!sleIssuer)
+        return tecNO_ISSUER;
+    if (!(sleIssuer->getFlags() & lsfAllowTokenLocking))
+        return tecNO_PERMISSION;
+
+    // If the account does not have a trustline to the issuer, return tecNO_LINE
+    auto const sleRippleState =
+        ctx.view.read(keylet::line(account, issuer, amount.getCurrency()));
+    if (!sleRippleState)
+        return tecNO_LINE;
+
+    STAmount const balance = (*sleRippleState)[sfBalance];
+
+    // If balance is positive, issuer must have higher address than account
```

Comment:
AFAIK "higher" / "lower" doesn't applicable to the accountID and doesn't have any meaning (except for uniqueness check).  What is this?